### ISSUE TITLE
Spot Perpetual Arbitrage - Does not look for min_divergence after second arbitrage

### ISSUE THREAD

**Issue by rapcmia** [CoinAlpha/hummingbot#3763]
**Describe the bug**
While checking the spot-perp arbitrage strategy, after the 2nd arbitrage it does not go back to min_divergence. It checks the min_convergence again and while it says it creates order to CLOSE position it creates OPEN position instead. 

**Steps To Reproduce**
1. Create spot-perpetual arbitrage strategy (Tested using ETH-USDT)
2. Monitor the bot until it triggers the min_divergence and then min_convergence
3. After the 2nd arbitrage is completed, the bot will still look for another min_convegence

On the screenshot below it look for another min_convergence after the 2nd arbitrage:
![image](https://user-images.githubusercontent.com/73840223/126318300-c4c48346-4d9a-4680-83c1-639c3354d1ec.png)

```
2021-07-20 19:39:18,999 - 14040 - hummingbot.strategy.spot_perpetual_arbitrage.spot_perpetual_arbitrage - INFO - Current spread: 0.05%, minimum divergence spread: 0.05%.
2021-07-20 19:39:18,999 - 14040 - hummingbot.strategy.spot_perpetual_arbitrage.spot_perpetual_arbitrage - INFO - Spread wide enough to execute first arbitrage!: 
Binance:  Sell  0.00600000 ETH at 1755.98000000 USDT. 
Binance_perpetual:  Buy  0.006 ETH at 1755.10 USDT. 

2021-07-20 19:39:19,000 - 14040 - hummingbot.strategy.spot_perpetual_arbitrage.spot_perpetual_arbitrage - INFO - Placing SELL order for 0.00600000 ETH at binance at 1755.98000000 price [clock=2021-07-20 11:39:19+00:00]
2021-07-20 19:39:19,000 - 14040 - hummingbot.strategy.spot_perpetual_arbitrage.spot_perpetual_arbitrage - INFO - Placing BUY order for 0.006 ETH at binance_perpetual at 1755.10 price to OPEN position. [clock=2021-07-20 11:39:19+00:00]
2021-07-20 19:39:19,097 - 14040 - hummingbot.connector.exchange.binance.binance_exchange - INFO - Created LIMIT SELL order x-XEKWYICX-SEHUT1626781159000747 for 0.00600000 ETH-USDT.
2021-07-20 19:39:19,097 - 14040 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626781159.0, "type": "OrderType.LIMIT", "trading_pair": "ETH-USDT", "amount": "0.00600000", "price": "1755.98000000", "order_id": "x-XEKWYICX-SEHUT1626781159000747", "exchange_order_id": "5075112106", "leverage": 1, "position": "NILL", "event_name": "SellOrderCreatedEvent", "event_source": "binance"}
2021-07-20 19:39:19,140 - 14040 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626781159.204, "order_id": "x-XEKWYICX-SEHUT1626781159000747", "trading_pair": "ETH-USDT", "trade_type": "TradeType.SELL", "order_type": "OrderType.LIMIT", "price": "1755.98000000", "amount": "0.00600000", "trade_fee": ["0", [["USDT", "0.01053588"]]], "exchange_trade_id": 523307821, "leverage": 1, "position": "NILL", "event_name": "OrderFilledEvent", "event_source": "binance"}
2021-07-20 19:39:19,177 - 14040 - hummingbot.connector.exchange.binance.binance_exchange - INFO - The market sell order x-XEKWYICX-SEHUT1626781159000747 has completed according to user stream.
2021-07-20 19:39:19,178 - 14040 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626781159.0, "order_id": "x-XEKWYICX-SEHUT1626781159000747", "base_asset": "ETH", "quote_asset": "USDT", "fee_asset": "USDT", "base_asset_amount": "0.00600000", "quote_asset_amount": "10.5358800000000000", "fee_amount": "0.01053588", "order_type": "OrderType.LIMIT", "exchange_order_id": null, "event_name": "SellOrderCompletedEvent", "event_source": "binance"}
2021-07-20 19:39:19,325 - 14040 - hummingbot.connector.derivative.binance_perpetual.binance_perpetual_derivative - INFO - Created market buy order x-3QreWesy-SEHUT1626781159000748 for 0.006 ETH-USDT.
2021-07-20 19:39:19,325 - 14040 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626781159.0, "type": "OrderType.MARKET", "trading_pair": "ETH-USDT", "amount": "0.006", "price": "1755.10", "order_id": "x-3QreWesy-SEHUT1626781159000748", "exchange_order_id": null, "leverage": 1, "position": "OPEN", "event_name": "BuyOrderCreatedEvent", "event_source": "binance_perpetual"}
2021-07-20 19:39:19,537 - 14040 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626781159.438, "order_id": "x-3QreWesy-SEHUT1626781159000748", "trading_pair": "ETH-USDT", "trade_type": "TradeType.BUY", "order_type": "OrderType.MARKET", "price": "1755.25", "amount": "0.006", "trade_fee": ["0.0004", []], "exchange_trade_id": 857044295, "leverage": 1, "position": "OPEN", "event_name": "OrderFilledEvent", "event_source": "binance_perpetual"}
2021-07-20 19:39:19,582 - 14040 - hummingbot.connector.derivative.binance_perpetual.binance_perpetual_derivative - INFO - The market TradeType.BUY order x-3QreWesy-SEHUT1626781159000748 has completed according to websocket delta.
2021-07-20 19:39:19,582 - 14040 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626781159.0, "order_id": "x-3QreWesy-SEHUT1626781159000748", "base_asset": "ETH", "quote_asset": "USDT", "fee_asset": "USDT", "base_asset_amount": "0.006", "quote_asset_amount": "10.53150", "fee_amount": "0.00421260", "order_type": "OrderType.MARKET", "exchange_order_id": null, "event_name": "BuyOrderCompletedEvent", "event_source": "binance_perpetual"}
2021-07-20 19:39:29,008 - 14040 - hummingbot.strategy.spot_perpetual_arbitrage.spot_perpetual_arbitrage - INFO - Current spread: 0.04%, minimum convergence spread: 0.01%.
2021-07-20 19:39:32,990 - 14040 - hummingbot.strategy.spot_perpetual_arbitrage.spot_perpetual_arbitrage - INFO - Current spread: 0.01%, minimum convergence spread: 0.01%.
2021-07-20 19:39:32,990 - 14040 - hummingbot.strategy.spot_perpetual_arbitrage.spot_perpetual_arbitrage - INFO - Spread low enough to execute second arbitrage!: 
Binance:  Buy  0.00600000 ETH at 1755.90000000 USDT. 
Binance_perpetual:  Sell  0.006 ETH at 1755.74 USDT. 

2021-07-20 19:39:32,991 - 14040 - hummingbot.strategy.spot_perpetual_arbitrage.spot_perpetual_arbitrage - INFO - Placing BUY order for 0.00600000 ETH at binance at 1755.90000000 price [clock=2021-07-20 11:39:33+00:00]
2021-07-20 19:39:32,991 - 14040 - hummingbot.strategy.spot_perpetual_arbitrage.spot_perpetual_arbitrage - INFO - Placing SELL order for 0.006 ETH at binance_perpetual at 1755.74 price to CLOSE position. [clock=2021-07-20 11:39:33+00:00]
2021-07-20 19:39:33,087 - 14040 - hummingbot.connector.exchange.binance.binance_exchange - INFO - Created LIMIT BUY order x-XEKWYICX-BEHUT1626781172991924 for 0.00600000 ETH-USDT.
2021-07-20 19:39:33,088 - 14040 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626781173.0, "type": "OrderType.LIMIT", "trading_pair": "ETH-USDT", "amount": "0.00600000", "price": "1755.90000000", "order_id": "x-XEKWYICX-BEHUT1626781172991924", "exchange_order_id": "5075113822", "leverage": 1, "position": "NILL", "event_name": "BuyOrderCreatedEvent", "event_source": "binance"}
2021-07-20 19:39:33,178 - 14040 - hummingbot.connector.derivative.binance_perpetual.binance_perpetual_derivative - INFO - Created market sell order x-3QreWesy-SEHUT1626781172991925 for 0.006 ETH-USDT.
2021-07-20 19:39:33,179 - 14040 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626781173.0, "type": "OrderType.MARKET", "trading_pair": "ETH-USDT", "amount": "0.006", "price": "1755.74", "order_id": "x-3QreWesy-SEHUT1626781172991925", "exchange_order_id": null, "leverage": 1, "position": "CLOSE", "event_name": "SellOrderCreatedEvent", "event_source": "binance_perpetual"}
2021-07-20 19:39:33,219 - 14040 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626781173.292, "order_id": "x-3QreWesy-SEHUT1626781172991925", "trading_pair": "ETH-USDT", "trade_type": "TradeType.SELL", "order_type": "OrderType.MARKET", "price": "1755.74", "amount": "0.003", "trade_fee": ["0.0004", []], "exchange_trade_id": 857044832, "leverage": 1, "position": "CLOSE", "event_name": "OrderFilledEvent", "event_source": "binance_perpetual"}
2021-07-20 19:39:33,312 - 14040 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626781173.292, "order_id": "x-3QreWesy-SEHUT1626781172991925", "trading_pair": "ETH-USDT", "trade_type": "TradeType.SELL", "order_type": "OrderType.MARKET", "price": "1755.74", "amount": "0.003", "trade_fee": ["0.0004", []], "exchange_trade_id": 857044833, "leverage": 1, "position": "CLOSE", "event_name": "OrderFilledEvent", "event_source": "binance_perpetual"}
2021-07-20 19:39:33,353 - 14040 - hummingbot.connector.derivative.binance_perpetual.binance_perpetual_derivative - INFO - The market TradeType.SELL order x-3QreWesy-SEHUT1626781172991925 has completed according to websocket delta.
2021-07-20 19:39:33,354 - 14040 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626781173.0, "order_id": "x-3QreWesy-SEHUT1626781172991925", "base_asset": "ETH", "quote_asset": "USDT", "fee_asset": "USDT", "base_asset_amount": "0.006", "quote_asset_amount": "10.53444", "fee_amount": "0.00421376", "order_type": "OrderType.MARKET", "exchange_order_id": null, "event_name": "SellOrderCompletedEvent", "event_source": "binance_perpetual"}
2021-07-20 19:40:09,266 - 14040 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626781209.355, "order_id": "x-XEKWYICX-BEHUT1626781172991924", "trading_pair": "ETH-USDT", "trade_type": "TradeType.BUY", "order_type": "OrderType.LIMIT", "price": "1755.90000000", "amount": "0.00600000", "trade_fee": ["0", [["ETH", "0.00000600"]]], "exchange_trade_id": 523308161, "leverage": 1, "position": "NILL", "event_name": "OrderFilledEvent", "event_source": "binance"}
2021-07-20 19:40:09,298 - 14040 - hummingbot.connector.exchange.binance.binance_exchange - INFO - The market buy order x-XEKWYICX-BEHUT1626781172991924 has completed according to user stream.
2021-07-20 19:40:09,298 - 14040 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626781209.0, "order_id": "x-XEKWYICX-BEHUT1626781172991924", "base_asset": "ETH", "quote_asset": "USDT", "fee_asset": "ETH", 
"base_asset_amount": "0.00600000", "quote_asset_amount": "10.5354000000000000", "fee_amount": "0.00000600", "order_type": "OrderType.LIMIT", "exchange_order_id": null, "event_name": "BuyOrderCompletedEvent", "event_source": "binance"}
2021-07-20 19:40:10,001 - 14040 - hummingbot.strategy.spot_perpetual_arbitrage.spot_perpetual_arbitrage - INFO - Current spread: 0.08%, minimum convergence spread: 0.01%

# I removed the unhandled error logs to localhost:5000 on timestamps : 2021-07-20 19:40:18,359 - 14040 - 

2021-07-20 19:40:20,993 - 14040 - hummingbot.strategy.spot_perpetual_arbitrage.spot_perpetual_arbitrage - INFO - Current spread: 0.04%, minimum convergence spread: 0.01%.
2021-07-20 19:40:31,993 - 14040 - hummingbot.strategy.spot_perpetual_arbitrage.spot_perpetual_arbitrage - INFO - Current spread: 0.06%, minimum convergence spread: 0.01%.
2021-07-20 19:40:41,000 - 14040 - hummingbot.strategy.spot_perpetual_arbitrage.spot_perpetual_arbitrage - INFO - Current spread: 0.01%, minimum convergence spread: 0.01%.
2021-07-20 19:40:41,000 - 14040 - hummingbot.strategy.spot_perpetual_arbitrage.spot_perpetual_arbitrage - INFO - Spread low enough to execute second arbitrage!: 
Binance:  Sell  0.00600000 ETH at 1756.48000000 USDT. 
Binance_perpetual:  Buy  0.006 ETH at 1756.65 USDT. 

2021-07-20 19:40:41,000 - 14040 - hummingbot.strategy.spot_perpetual_arbitrage.spot_perpetual_arbitrage - INFO - Placing SELL order for 0.00600000 ETH at binance at 1756.48000000 price [clock=2021-07-20 11:40:41+00:00]
2021-07-20 19:40:41,001 - 14040 - hummingbot.strategy.spot_perpetual_arbitrage.spot_perpetual_arbitrage - INFO - Placing BUY order for 0.006 ETH at binance_perpetual at 1756.65 price to CLOSE position. [clock=2021-07-20 11:40:41+00:00]
```

**Release version**
dev-0.42.0

**Attachments**
[spot-perp-arb.zip](https://github.com/CoinAlpha/hummingbot/files/6848176/spot-perp-arb.zip)

*WARNING: Do NOT publish any exchange API keys or your wallet's private key. Whoever has access to them may steal your assets!*

**Optional: your discord username:**

**Comment by JK5768** [comment]
I am having exactly the same issue.

**Comment by kienzli** [comment]
I'm having the same issue:
[conf_spot_perpetual_arbitrage_1.yml.txt](https://github.com/CoinAlpha/hummingbot/files/7006421/conf_spot_perpetual_arbitrage_1.yml.txt)
[logs_conf_perpetual_arbitrage_1.log](https://github.com/CoinAlpha/hummingbot/files/7006422/logs_conf_perpetual_arbitrage_1.log)

**Comment by kienzli** [comment]
> I'm having the same issue:
> [conf_spot_perpetual_arbitrage_1.yml.txt](https://github.com/CoinAlpha/hummingbot/files/7006421/conf_spot_perpetual_arbitrage_1.yml.txt)
> [logs_conf_perpetual_arbitrage_1.log](https://github.com/CoinAlpha/hummingbot/files/7006422/logs_conf_perpetual_arbitrage_1.log)

=> on version 0.42.0

**Comment by torsteinbakke** [comment]
me too!

**Comment by kienzli** [comment]
> **Describe the bug**
> While checking the spot-perp arbitrage strategy, after the 2nd arbitrage it does not go back to min_divergence. It checks the min_convergence again and while it says it creates order to CLOSE position it creates OPEN position instead.
> 
> **Steps To Reproduce**
> 
>     1. Create spot-perpetual arbitrage strategy (Tested using ETH-USDT)
> 
>     2. Monitor the bot until it triggers the min_divergence and then min_convergence
> 
>     3. After the 2nd arbitrage is completed, the bot will still look for another min_convegence
> 
> 
> On the screenshot below it look for another min_convergence after the 2nd arbitrage:
> ![image](https://user-images.githubusercontent.com/73840223/126318300-c4c48346-4d9a-4680-83c1-639c3354d1ec.png)
> 
> ```
> 2021-07-20 19:39:18,999 - 14040 - hummingbot.strategy.spot_perpetual_arbitrage.spot_perpetual_arbitrage - INFO - Current spread: 0.05%, minimum divergence spread: 0.05%.
> 2021-07-20 19:39:18,999 - 14040 - hummingbot.strategy.spot_perpetual_arbitrage.spot_perpetual_arbitrage - INFO - Spread wide enough to execute first arbitrage!: 
> Binance:  Sell  0.00600000 ETH at 1755.98000000 USDT. 
> Binance_perpetual:  Buy  0.006 ETH at 1755.10 USDT. 
> 
> 2021-07-20 19:39:19,000 - 14040 - hummingbot.strategy.spot_perpetual_arbitrage.spot_perpetual_arbitrage - INFO - Placing SELL order for 0.00600000 ETH at binance at 1755.98000000 price [clock=2021-07-20 11:39:19+00:00]
> 2021-07-20 19:39:19,000 - 14040 - hummingbot.strategy.spot_perpetual_arbitrage.spot_perpetual_arbitrage - INFO - Placing BUY order for 0.006 ETH at binance_perpetual at 1755.10 price to OPEN position. [clock=2021-07-20 11:39:19+00:00]
> 2021-07-20 19:39:19,097 - 14040 - hummingbot.connector.exchange.binance.binance_exchange - INFO - Created LIMIT SELL order x-XEKWYICX-SEHUT1626781159000747 for 0.00600000 ETH-USDT.
> 2021-07-20 19:39:19,097 - 14040 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626781159.0, "type": "OrderType.LIMIT", "trading_pair": "ETH-USDT", "amount": "0.00600000", "price": "1755.98000000", "order_id": "x-XEKWYICX-SEHUT1626781159000747", "exchange_order_id": "5075112106", "leverage": 1, "position": "NILL", "event_name": "SellOrderCreatedEvent", "event_source": "binance"}
> 2021-07-20 19:39:19,140 - 14040 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626781159.204, "order_id": "x-XEKWYICX-SEHUT1626781159000747", "trading_pair": "ETH-USDT", "trade_type": "TradeType.SELL", "order_type": "OrderType.LIMIT", "price": "1755.98000000", "amount": "0.00600000", "trade_fee": ["0", [["USDT", "0.01053588"]]], "exchange_trade_id": 523307821, "leverage": 1, "position": "NILL", "event_name": "OrderFilledEvent", "event_source": "binance"}
> 2021-07-20 19:39:19,177 - 14040 - hummingbot.connector.exchange.binance.binance_exchange - INFO - The market sell order x-XEKWYICX-SEHUT1626781159000747 has completed according to user stream.
> 2021-07-20 19:39:19,178 - 14040 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626781159.0, "order_id": "x-XEKWYICX-SEHUT1626781159000747", "base_asset": "ETH", "quote_asset": "USDT", "fee_asset": "USDT", "base_asset_amount": "0.00600000", "quote_asset_amount": "10.5358800000000000", "fee_amount": "0.01053588", "order_type": "OrderType.LIMIT", "exchange_order_id": null, "event_name": "SellOrderCompletedEvent", "event_source": "binance"}
> 2021-07-20 19:39:19,325 - 14040 - hummingbot.connector.derivative.binance_perpetual.binance_perpetual_derivative - INFO - Created market buy order x-3QreWesy-SEHUT1626781159000748 for 0.006 ETH-USDT.
> 2021-07-20 19:39:19,325 - 14040 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626781159.0, "type": "OrderType.MARKET", "trading_pair": "ETH-USDT", "amount": "0.006", "price": "1755.10", "order_id": "x-3QreWesy-SEHUT1626781159000748", "exchange_order_id": null, "leverage": 1, "position": "OPEN", "event_name": "BuyOrderCreatedEvent", "event_source": "binance_perpetual"}
> 2021-07-20 19:39:19,537 - 14040 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626781159.438, "order_id": "x-3QreWesy-SEHUT1626781159000748", "trading_pair": "ETH-USDT", "trade_type": "TradeType.BUY", "order_type": "OrderType.MARKET", "price": "1755.25", "amount": "0.006", "trade_fee": ["0.0004", []], "exchange_trade_id": 857044295, "leverage": 1, "position": "OPEN", "event_name": "OrderFilledEvent", "event_source": "binance_perpetual"}
> 2021-07-20 19:39:19,582 - 14040 - hummingbot.connector.derivative.binance_perpetual.binance_perpetual_derivative - INFO - The market TradeType.BUY order x-3QreWesy-SEHUT1626781159000748 has completed according to websocket delta.
> 2021-07-20 19:39:19,582 - 14040 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626781159.0, "order_id": "x-3QreWesy-SEHUT1626781159000748", "base_asset": "ETH", "quote_asset": "USDT", "fee_asset": "USDT", "base_asset_amount": "0.006", "quote_asset_amount": "10.53150", "fee_amount": "0.00421260", "order_type": "OrderType.MARKET", "exchange_order_id": null, "event_name": "BuyOrderCompletedEvent", "event_source": "binance_perpetual"}
> 2021-07-20 19:39:29,008 - 14040 - hummingbot.strategy.spot_perpetual_arbitrage.spot_perpetual_arbitrage - INFO - Current spread: 0.04%, minimum convergence spread: 0.01%.
> 2021-07-20 19:39:32,990 - 14040 - hummingbot.strategy.spot_perpetual_arbitrage.spot_perpetual_arbitrage - INFO - Current spread: 0.01%, minimum convergence spread: 0.01%.
> 2021-07-20 19:39:32,990 - 14040 - hummingbot.strategy.spot_perpetual_arbitrage.spot_perpetual_arbitrage - INFO - Spread low enough to execute second arbitrage!: 
> Binance:  Buy  0.00600000 ETH at 1755.90000000 USDT. 
> Binance_perpetual:  Sell  0.006 ETH at 1755.74 USDT. 
> 
> 2021-07-20 19:39:32,991 - 14040 - hummingbot.strategy.spot_perpetual_arbitrage.spot_perpetual_arbitrage - INFO - Placing BUY order for 0.00600000 ETH at binance at 1755.90000000 price [clock=2021-07-20 11:39:33+00:00]
> 2021-07-20 19:39:32,991 - 14040 - hummingbot.strategy.spot_perpetual_arbitrage.spot_perpetual_arbitrage - INFO - Placing SELL order for 0.006 ETH at binance_perpetual at 1755.74 price to CLOSE position. [clock=2021-07-20 11:39:33+00:00]
> 2021-07-20 19:39:33,087 - 14040 - hummingbot.connector.exchange.binance.binance_exchange - INFO - Created LIMIT BUY order x-XEKWYICX-BEHUT1626781172991924 for 0.00600000 ETH-USDT.
> 2021-07-20 19:39:33,088 - 14040 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626781173.0, "type": "OrderType.LIMIT", "trading_pair": "ETH-USDT", "amount": "0.00600000", "price": "1755.90000000", "order_id": "x-XEKWYICX-BEHUT1626781172991924", "exchange_order_id": "5075113822", "leverage": 1, "position": "NILL", "event_name": "BuyOrderCreatedEvent", "event_source": "binance"}
> 2021-07-20 19:39:33,178 - 14040 - hummingbot.connector.derivative.binance_perpetual.binance_perpetual_derivative - INFO - Created market sell order x-3QreWesy-SEHUT1626781172991925 for 0.006 ETH-USDT.
> 2021-07-20 19:39:33,179 - 14040 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626781173.0, "type": "OrderType.MARKET", "trading_pair": "ETH-USDT", "amount": "0.006", "price": "1755.74", "order_id": "x-3QreWesy-SEHUT1626781172991925", "exchange_order_id": null, "leverage": 1, "position": "CLOSE", "event_name": "SellOrderCreatedEvent", "event_source": "binance_perpetual"}
> 2021-07-20 19:39:33,219 - 14040 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626781173.292, "order_id": "x-3QreWesy-SEHUT1626781172991925", "trading_pair": "ETH-USDT", "trade_type": "TradeType.SELL", "order_type": "OrderType.MARKET", "price": "1755.74", "amount": "0.003", "trade_fee": ["0.0004", []], "exchange_trade_id": 857044832, "leverage": 1, "position": "CLOSE", "event_name": "OrderFilledEvent", "event_source": "binance_perpetual"}
> 2021-07-20 19:39:33,312 - 14040 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626781173.292, "order_id": "x-3QreWesy-SEHUT1626781172991925", "trading_pair": "ETH-USDT", "trade_type": "TradeType.SELL", "order_type": "OrderType.MARKET", "price": "1755.74", "amount": "0.003", "trade_fee": ["0.0004", []], "exchange_trade_id": 857044833, "leverage": 1, "position": "CLOSE", "event_name": "OrderFilledEvent", "event_source": "binance_perpetual"}
> 2021-07-20 19:39:33,353 - 14040 - hummingbot.connector.derivative.binance_perpetual.binance_perpetual_derivative - INFO - The market TradeType.SELL order x-3QreWesy-SEHUT1626781172991925 has completed according to websocket delta.
> 2021-07-20 19:39:33,354 - 14040 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626781173.0, "order_id": "x-3QreWesy-SEHUT1626781172991925", "base_asset": "ETH", "quote_asset": "USDT", "fee_asset": "USDT", "base_asset_amount": "0.006", "quote_asset_amount": "10.53444", "fee_amount": "0.00421376", "order_type": "OrderType.MARKET", "exchange_order_id": null, "event_name": "SellOrderCompletedEvent", "event_source": "binance_perpetual"}
> 2021-07-20 19:40:09,266 - 14040 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626781209.355, "order_id": "x-XEKWYICX-BEHUT1626781172991924", "trading_pair": "ETH-USDT", "trade_type": "TradeType.BUY", "order_type": "OrderType.LIMIT", "price": "1755.90000000", "amount": "0.00600000", "trade_fee": ["0", [["ETH", "0.00000600"]]], "exchange_trade_id": 523308161, "leverage": 1, "position": "NILL", "event_name": "OrderFilledEvent", "event_source": "binance"}
> 2021-07-20 19:40:09,298 - 14040 - hummingbot.connector.exchange.binance.binance_exchange - INFO - The market buy order x-XEKWYICX-BEHUT1626781172991924 has completed according to user stream.
> 2021-07-20 19:40:09,298 - 14040 - hummingbot.core.event.event_reporter - EVENT_LOG - {"timestamp": 1626781209.0, "order_id": "x-XEKWYICX-BEHUT1626781172991924", "base_asset": "ETH", "quote_asset": "USDT", "fee_asset": "ETH", 
> "base_asset_amount": "0.00600000", "quote_asset_amount": "10.5354000000000000", "fee_amount": "0.00000600", "order_type": "OrderType.LIMIT", "exchange_order_id": null, "event_name": "BuyOrderCompletedEvent", "event_source": "binance"}
> 2021-07-20 19:40:10,001 - 14040 - hummingbot.strategy.spot_perpetual_arbitrage.spot_perpetual_arbitrage - INFO - Current spread: 0.08%, minimum convergence spread: 0.01%
> 
> # I removed the unhandled error logs to localhost:5000 on timestamps : 2021-07-20 19:40:18,359 - 14040 - 
> 
> 2021-07-20 19:40:20,993 - 14040 - hummingbot.strategy.spot_perpetual_arbitrage.spot_perpetual_arbitrage - INFO - Current spread: 0.04%, minimum convergence spread: 0.01%.
> 2021-07-20 19:40:31,993 - 14040 - hummingbot.strategy.spot_perpetual_arbitrage.spot_perpetual_arbitrage - INFO - Current spread: 0.06%, minimum convergence spread: 0.01%.
> 2021-07-20 19:40:41,000 - 14040 - hummingbot.strategy.spot_perpetual_arbitrage.spot_perpetual_arbitrage - INFO - Current spread: 0.01%, minimum convergence spread: 0.01%.
> 2021-07-20 19:40:41,000 - 14040 - hummingbot.strategy.spot_perpetual_arbitrage.spot_perpetual_arbitrage - INFO - Spread low enough to execute second arbitrage!: 
> Binance:  Sell  0.00600000 ETH at 1756.48000000 USDT. 
> Binance_perpetual:  Buy  0.006 ETH at 1756.65 USDT. 
> 
> 2021-07-20 19:40:41,000 - 14040 - hummingbot.strategy.spot_perpetual_arbitrage.spot_perpetual_arbitrage - INFO - Placing SELL order for 0.00600000 ETH at binance at 1756.48000000 price [clock=2021-07-20 11:40:41+00:00]
> 2021-07-20 19:40:41,001 - 14040 - hummingbot.strategy.spot_perpetual_arbitrage.spot_perpetual_arbitrage - INFO - Placing BUY order for 0.006 ETH at binance_perpetual at 1756.65 price to CLOSE position. [clock=2021-07-20 11:40:41+00:00]
> ```
> 
> **Release version**
> dev-0.42.0
> 
> **Attachments**
> [spot-perp-arb.zip](https://github.com/CoinAlpha/hummingbot/files/6848176/spot-perp-arb.zip)
> 
> _WARNING: Do NOT publish any exchange API keys or your wallet's private key. Whoever has access to them may steal your assets!_
> 
> **Optional: your discord username:**

the deriv_position is also still open, which could possibly lead to not switching back..

**Comment by zmei705** [comment]
[logs_conf_spot_perpetual_arbitrage_15.log](https://github.com/CoinAlpha/hummingbot/files/7027300/logs_conf_spot_perpetual_arbitrage_15.log)
I am having exactly the same issue.

**Comment by kevin9733** [comment]
I am having exactly the same issue.

**Comment by dennisocana** [comment]
Fixed by @kienzli in version 0.43 release.
